src/include/btree_inline.h

```diff
@@ -874,151 +874,171 @@ __wt_ref_key_clear(WT_REF *ref)
- * (we could even tune for specific workloads).  Generally, the fields
- * are larger than the anticipated values being stored (512B keys, 8KB
- * values, 1MB pages), hopefully that won't be necessary.
- *
- * This function returns a list of things about the key (instantiation
- * reference, cell reference and key/length pair).  Our callers know
- * the order in which we look things up and the information returned;
- * for example, the cell will never be returned if we are working with
- * an on-page key.
- */
+    /*
+     * A row-store leaf page key is in one of two places: if instantiated, the WT_ROW pointer
+     * references a WT_IKEY structure, otherwise, it references an on-page item. Further, on-page
+     * items are in one of two states: if the key is a simple key (not an overflow key, which is
+     * likely), the key's offset, size and prefix is encoded in the 8B of pointer. Otherwise, the
+     * offset is to the key's on-page cell.
+     *
+     * This function returns information from set of things about the key (WT_IKEY reference, cell
+     * reference and/or key/length/prefix triplet). Our callers know the order we resolve items and
+     * what information will be returned. Specifically, the caller gets a key (in the form of a
+     * pointer to the bytes, a length and a prefix length in all cases where we can get it without
+     * unpacking a cell), plus an optional WT_IKEY reference, and in all cases, a pointer to the
+     * on-page cell. Our caller's test is generally if there is a returned key or not, falling back
+     * to the returned cell.
+     *
+     * Now the magic: allocated memory must be aligned to store any standard type and we expect some
+     * standard type to require at least quad-byte alignment, so allocated memory should have two
+     * clear low-order bits. On-page objects consist of an offset/length pair and a prefix in the
+     * case of a key: the maximum page size is 29 bits (512MB), the remaining bits hold the key or
+     * value location and bytes. This breaks if allocated memory isn't aligned, of course.
+     *
+     * In this specific case, we use bit 0x01 to mark an on-page cell, bit 0x02 to mark an on-page
+     * key, 0x03 to mark an on-page key/value pair, otherwise it's a WT_IKEY reference. The bit
+     * pattern for on-page cells is:
+     *
+     *	29 bits		offset of the key's cell (512MB)
+     *	 2 bits		0x01 flag
+     *
+     * The on-page cell is our fallback: if a key or value won't fit into our encoding (unlikely,
+     * but possible), we fall back to using a cell reference, which obviously has enough room for
+     * all possible values.
+     *
+     * The next encoding is for on-page keys:
+     *
+     *	19 bits		key length (512KB)
+     *   6 bits		offset of the key's bytes from the key's cell (32B)
+     *   8 bits		key prefix (256B)
+     *	29 bits		offset of the key's cell (512MB)
+     *	 2 bits		0x02 flag
```

Comment:
There's a formatting issue, probably tab/space on the two new lines with `6 bits` and `8 bits`.

Reply:
It's just an issue with the diff, it doesn't matter when you're editing the file or reading through it.

Anyway, I went through and regularized all of the lines to only use spaces in 31e0760fc.